### PR TITLE
Transfer to new drone

### PR DESCRIPTION
**Pull request by tasharnvb**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ workspace:
 
 pipeline:
   build_binary:
-    image: golang:1.8.1
```
Comment (jaykeshur) [inline .drone.yml]: probably shouldn't change this, unless you've tested it compiles and works properly

Comment (tasharnvb) [reply .drone.yml]: i haven't... i'll change it back

File: .drone.yml [hunk 2]
```diff
@@ -15,32 +15,36 @@ pipeline:
       - GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -o bin/kube-cfn-signal_windows_amd64.exe
 
   build_docker_image:
-    image: docker:1.12
+    image: docker:17.12.0
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
       - docker build -t image .
```
Comment (marcinc) [inline .drone.yml]: you probably want to build that images with $${DRONE_COMMIT_SHA}

File: .drone.yml [hunk 3]
```diff
@@ -15,32 +15,36 @@ pipeline:
...
+    secrets:
+      - docker_password
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker login -u="ukhomeofficedigital+kube_cfn_signal" -p=${DOCKER_PASSWORD} quay.io
+      - docker login -u="ukhomeofficedigital+kube_cfn_signal" -p=$${DOCKER_PASSWORD} quay.io
       - docker tag image quay.io/ukhomeofficedigital/kube-cfn-signal:latest
       - docker push quay.io/ukhomeofficedigital/kube-cfn-signal:latest
```
Comment (marcinc) [inline .drone.yml]: and here you would tag image with both $${DRONE_COMMIT_SHA} and latest.

File: .drone.yml (hunk 2 above)
Comment (tasharnvb) [reply .drone.yml]: will do

File: .drone.yml [hunk 4]
```diff
@@ -15,32 +15,38 @@ pipeline:
...
 
   build_docker_image:
-    image: docker:1.12
+    image: docker:17.12.0
     environment:
-      - DOCKER_HOST=tcp://127.0.0.1:2375
+      - DOCKER_HOST=tcp://172.17.0.1:2375
     commands:
-      - docker build -t image .
+      - docker build -t image:$${DRONE_COMMIT_SHA} .
```
Comment (marcinc) [inline .drone.yml]: Probably worth changing `image` to `kube-cfn-signal` for clarity.

**Review by marcinc — APPROVED** [review]
lgtm

**Pull request merged**